### ISSUE TITLE
[BUG][mthreads] llc crash (SIGSEGV) in MTGPU PreRA Optimization at compile time

### ISSUE THREAD

**Issue by O1iveira7** [FlagTree/flagtree#980]
## Environment
  - Triton 3.6.0 (FlagTree mthreads3.6), MUSA backend, MTT S5000
  - torch_musa 2.7.1, MUSA 4.3.5, Python 3.10
## Symptom
  Compiling certain Triton kernels crashes the host-side compiler `llc` with a
  deterministic SIGSEGV (exit -11) inside pass **MTGPU PreRA Optimization**.
  The Triton frontend (TTIR/TTGIR/LLIR) produces IR normally The crash is triggered only by the*combination* of three opaque-branch chains that are dead code at runtime
  (conditions loaded from device memory):

  - **B1**: sin/cos chain + once-loop with masked store
  - **B2**: f16 → bf16 → f32 conversion chain + `tl.sum` reduction
  - **B4**: `swizzle2d` + `multiple_of`/`max_contiguous`/`max_constancy` + `tl.assume`

 removing **any one** of B1/B2/B4 makes the crash disappear.

**FlagTree mthreads3.6 output:**
>  llc SIGSEGV -11: `llc` failed with error code -11
> exit=0

**Triton3.7 RTX3070 output:**
>no crash: kernel ran
> output = [1.0, 3.0, 6.0, 10.0]

 ## Minimal reproducer
 ### Difference between the MTT S5000 and NVIDIA RTX3070
```
      if not torch.musa.is_available():
      ...

      output = torch.zeros(4, dtype=torch.float32, device="musa")
      ...
          torch.musa.synchronize()

```
```
    if not torch.cuda.is_available():
   ...
DEVICE = "cuda"
    data = torch.tensor(DATA, dtype=torch.float32)
    output = torch.zeros(4, dtype=torch.float32, device=DEVICE)
...
        solve(data.to(DEVICE), output, N)
        torch.cuda.synchronize()
...
```
### reproducer
```
● #!/usr/bin/env python3
  # -*- coding: utf-8 -*-
  """
=====================================================================
  Reproducer for the reduced variant (source-level delta-reduce result
  { B1, B2, B4 }, see evidence/delta_reduction_table.txt): skeleton +
  B1 sin/cos chain, B2 f16->bf16 chain, B4 swizzle2d/assume chain.
  Removing any one of B1/B2/B4 makes the crash disappear.

  Judge: compiling @partial_cum_sum_min raises the llc SIGSEGV RuntimeError
  (error code -11, pass 'MTGPU PreRA Optimization'). Exit 0 = reproduced.

  Usage: MUSA_VISIBLE_DEVICES=0 python3 repro_000003_minimal.py
  """
  import sys

  import torch
  import triton
  import triton.language as tl

  DATA = [1.0, 2.0, 3.0, 4.0]
  N = 4


  @triton.jit
  def partial_cum_sum_min(
      _fz_switch_ptr,
      _fz_switch_output_ptr,
      data_ptr,
      output_ptr,
      grid_sum_ptr,
      n,
      BLOCK_SIZE: tl.constexpr,
  ):
      opaque_values = tl.load(_fz_switch_ptr + tl.arange(0, 2))
      (opaque_zero, opaque_one) = tl.split(opaque_values)
      opaque_true = opaque_zero < opaque_one
      opaque_false = opaque_zero > opaque_one

      # B1: sin/cos chain + once-loop scratch store
      if opaque_true:
          _fz_a8_sub_value = opaque_one - opaque_zero
          _fz_a8_add_value = opaque_zero + opaque_one
          _fz_a8_mul_value = (_fz_a8_add_value + 1) * (_fz_a8_sub_value + 1)
          _fz_a8_float_value = _fz_a8_mul_value.to(tl.float32)
          _fz_a8_div_value = _fz_a8_float_value / (opaque_one + 1)
          _fz_a8_row_value = tl.expand_dims(opaque_values + _fz_a8_div_value, 0)
          _fz_a8_transposed_value = tl.trans(_fz_a8_row_value)
          _fz_a8_sin_value = tl.sin(_fz_a8_transposed_value)
          _fz_a8_cos_value = tl.cos(_fz_a8_sin_value)
          for _fz_a11_loop_once in range(0, 1):
              if opaque_true:
                  tl.store(
                      _fz_switch_output_ptr + 14 + tl.arange(0, 2),
                      tl.ravel(_fz_a8_cos_value),
                      mask=(tl.program_id(axis=0) == 0) & (tl.program_id(axis=1) == 0) & (tl.program_id(axis=2) == 0),
                  )
              else:
                  pass

      # B2: f16 -> bf16 chain
      if opaque_false:
          _fz_a18_f16_x = tl.arange(0, 8).to(tl.float16) + opaque_one.to(tl.float16)
          _fz_a18_f16_y = _fz_a18_f16_x * 1.5 - 0.5
          _fz_a18_f16_z = _fz_a18_f16_y.to(tl.float32) + _fz_a18_f16_x.to(tl.float32)
          _fz_a18_f16_w = _fz_a18_f16_z.to(tl.bfloat16).to(tl.float32)
          _fz_a18_tpl_result = tl.sum(_fz_a18_f16_w / 4.0) + opaque_values.to(tl.float32)
          tl.store(
              _fz_switch_output_ptr + 34 + tl.arange(0, 2),
              _fz_a18_tpl_result,
              mask=(tl.program_id(axis=0) == 0) & (tl.program_id(axis=1) == 0) & (tl.program_id(axis=2) == 0),
          )

      # B4: swizzle2d + assume chain
      if opaque_false:
          _fz_a19_cre_a = tl.full((4, 4), 2, tl.int32) + opaque_one
          _fz_a19_cre_z = tl.zeros((4, 4), tl.float32)
          _fz_a19_cre_m = tl.multiple_of(tl.arange(0, 16), 16)
          _fz_a19_cre_c = tl.max_contiguous(_fz_a19_cre_m, 8)
          _fz_a19_cre_k = tl.max_constancy(_fz_a19_cre_c, 8)
          _fz_a19_cre_g = tl.reshape(tl.arange(0, 16), (4, 4)) + opaque_one
          (_fz_a19_cre_si, _fz_a19_cre_sj) = tl.swizzle2d(_fz_a19_cre_g // 4, _fz_a19_cre_g % 4, 4, 4, 2)
          _fz_a19_cre_w = tl.where((_fz_a19_cre_si + _fz_a19_cre_sj) % 2 == 0, _fz_a19_cre_a, _fz_a19_cre_a + 1)
          tl.assume(opaque_true)
          _fz_a19_cre_f = tl.sum(tl.ravel(_fz_a19_cre_w.to(tl.float32) + _fz_a19_cre_z) +
  tl.ravel(_fz_a19_cre_k.to(tl.float32)))
          _fz_a19_tpl_result = _fz_a19_cre_f + opaque_values.to(tl.float32)
          tl.store(
              _fz_switch_output_ptr + 36 + tl.arange(0, 2),
              _fz_a19_tpl_result,
              mask=(tl.program_id(axis=0) == 0) & (tl.program_id(axis=1) == 0) & (tl.program_id(axis=2) == 0),
          )
          return

      PID = tl.program_id(0)
      if opaque_true:
          offset = PID * BLOCK_SIZE + tl.arange(0, BLOCK_SIZE)
      else:
          offset = PID * BLOCK_SIZE + tl.arange(0, BLOCK_SIZE)
      mask = offset < n
      block = tl.load(data_ptr + offset, mask=mask, other=0.0)
      local_cum_sum = tl.cumsum(block, axis=0)
      tl.store(output_ptr + offset, local_cum_sum, mask=mask)
      max_sum = tl.sum(block)
      tl.store(grid_sum_ptr + PID, max_sum)


  def solve(data: torch.Tensor, output: torch.Tensor, n: int):
      switch_ptr = torch.tensor([0, 1], dtype=torch.int32, device=data.device)
      switch_output_ptr = torch.empty(400, dtype=torch.float32, device=data.device)
      BLOCK_SIZE = 1024
      num_blocks = triton.cdiv(n, BLOCK_SIZE)
      grid = (num_blocks,)
      grid_sum = torch.zeros(num_blocks, device=data.device, dtype=data.dtype)
      partial_cum_sum_min[grid](switch_ptr, switch_output_ptr, data, output, grid_sum, n, BLOCK_SIZE=BLOCK_SIZE)


  def main() -> int:
      if not torch.musa.is_available():
          print("MUSA is unavailable", file=sys.stderr)
          return 2

      data = torch.tensor(DATA, dtype=torch.float32)
      output = torch.zeros(4, dtype=torch.float32, device="musa")
      try:
          solve(data.to("musa"), output, N)
          torch.musa.synchronize()
      except Exception as e:
          msg = str(e)
          if "llc" in msg and "-11" in msg:
              print(f"llc SIGSEGV -11: {msg.splitlines()[0][:80]}")
              return 0
          print(f"FAIL: {type(e).__name__}: {msg.splitlines()[0][:80]}")
          return 1
      print("no crash: kernel ran")
      print(f"  output = {output.cpu().tolist()}")
      return 1


  if __name__ == "__main__":
      raise SystemExit(main())
```

**Comment by zhzhcookie** [comment]
Please try `flagtree===0.6.2a3+mthreads3.6`, or build flagtree from source.
